src/System.Private.CoreLib/shared/System/Activator.cs

```diff
@@ -18,36 +19,36 @@ public static partial class Activator
 
         [DebuggerHidden]
         [DebuggerStepThrough]
-        public static object CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, object[] args, CultureInfo culture) =>
+        public static object? CreateInstance(Type type, BindingFlags bindingAttr, Binder? binder, object?[]? args, CultureInfo? culture) =>
```

Comment:
Why did you annotate all `CreateInstance` methods' return value as nullable?

Reply:
`CreateInstance` will return null for `Nullable<T>` types, for example, try:

`Console.WriteLine(Activator.CreateInstance(typeof(int?)) == null);`

Reply:
Oh, okay, I forgot about nullable types boxing. So only the generic version of CreateInstance will provide us with not null value for classes using unknown nullability of generic-parameter.
I'm a bit disappointed, that `CreateInstance<SomeClass?>` still returns SomeClass? instead of `SomeClass`, but to handle this we need to add two different methods with struct and class constraints and mark second with [NotNull], but this sounds as overengineering.
Anyway, thanks for explanation